### PR TITLE
feat(stepfunctions-tasks): support FLEX execution class for GlueStartJobRun

### LINKED ISSUE #30533: stepfunctions-tasks: support FLEX execution class for GlusStartJobRun
### Describe the feature

Glue Job supports [FLEX execution class](https://aws.amazon.com/about-aws/whats-new/2022/08/aws-glue-supports-flex-execution-option/?nc1=h_ls).

This feature can be used by setting `ExecutionClass` in `StartJobRun`.

But current [GlusStartJobRun](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions_tasks.GlueStartJobRun.html) class does not support it.

### Use Case

To reduce job running cost by using FLEX class.

### Proposed Solution

Add `executionClass` property.

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.143.0

### Environment details (OS name and version, etc.)

all

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #30533 

### Reason for this change
To support `FLEX` execution class for `GlusStartJobRun`.
By using `FLEX` class, you can reduce the cost of running Glue job.

### Description of changes
Add `execution class` to the `GlusStartJobRun` class.

### Description of how you validated changes
Add unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts [hunk 1]
```diff
@@ -167,35 +175,52 @@ export enum WorkerType {
...
+  Z_2X = 'Z.2X',
+}
+
+/**
+ * The excecution class of the job.
+ */
+export enum ExecutionClass {
+  /**
+   * The flexible execution class is appropriate for time-insensitive jobs whose start and completion times may vary.
+   * Only jobs with AWS Glue version 3.0 and above and command type glueetl will be allowed to set ExecutionClass to FLEX.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts:217]:
```suggestion
   * Only jobs with AWS Glue version 3.0 and above and command type `glueetl` will be allowed to set `ExecutionClass` to `FLEX`.
```

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/README.md [hunk 2]
```diff
@@ -1142,6 +1142,15 @@ new tasks.GlueStartJobRun(this, 'Task', {
 });
 ```
 
+Also you can choose execution class by settint the `executionClass` property.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/README.md:1145]:
```suggestion
You can choose the execution class by setting the `executionClass` property.
```

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts [hunk 3]
```diff
@@ -167,35 +175,52 @@ export enum WorkerType {
...
    * Each worker maps to 8 DPU (32 vCPU, 128 GB of memory, 512 GB disk), and provides 1 executor per worker. We recommend this worker type for jobs whose workloads contain your most demanding transforms, aggregations, joins, and queries. This worker type is available only for AWS Glue version 3.0 or later jobs.
    */
-  G_8X ='G.8X',
+  G_8X = 'G.8X',
 
   /**
    * Each worker maps to 0.25 DPU (2 vCPU, 4 GB of memory, 64 GB disk), and provides 1 executor per worker. Suitable for low volume streaming jobs.
    */
-  G_025X ='G.025X',
+  G_025X = 'G.025X',
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts:203]: Can you please move `G_025X` between `STANDARD` and `G_1X`?
It seems a bit out of place here.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts (hunk 3 above)
Comment (mazyu36) [reply packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts]: @lpizzinidev 
Thanks. I've changed the order.

**Review by GavinZZ — APPROVED** [review]
LGTM, thanks for making contribution.

**Comment by mazyu36** [comment]
@Mergifyio update

**Comment by mazyu36** [comment]
@Mergifyio update

**Pull request merged**